operator panics during the `replace` procedure: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Slack](https://scylladb-users-slackin.herokuapp.com/).
3. Did you already search the existing open issues for anything similar? -->

**Describe the bug**
During the `replace Scylla member` procedure on the step where the pod Scylla gets deleted and substep "deletion of the endpoint"
scylla operator crashed.
SCT logs:
```
2023-06-11 23:17:19,450 f:base.py         l:142  c:LocalCmdRunner       p:DEBUG > Command "kubectl --cache-dir=/home/ubuntu/sct-results/20230611-210029-332173/.kube/http-cache --namespace=scylla wait --timeout=21600s --for=delete pod sct-cluster-us-east1-b-us-east1-2" finished with status 0
...
2023-06-11 23:17:19,486 f:__init__.py     l:793  c:sdcm.utils.k8s       p:DEBUG > Processing following line: \
  {"type":"MODIFIED","object":{"kind":"Endpoints","apiVersion":"v1","metadata":{ \
    "name":"sct-cluster-us-east1-b-us-east1-2","namespace":"scylla","uid":"ab9b398c-da9e-4ae2-b154-84410f42ad25","resourceVersion":"42603","creationTimestamp":"2023-06-11T23:10:18Z","labels":{"app":"scylla","app.kubernetes.io/managed-by":"scylla-operator","app.kubernetes.io/name":"scylla","scylla-operator.scylladb.com/scylla-service-type":"member","scylla/cluster":"sct-cluster","scylla/datacenter":"us-east1-b","scylla/decommissioned":"true","scylla/rack":"us-east1","scylla/replace":"172.20.26.177"},"managedFields":[{"manager":"kube-controller-manager","operation":"Update","apiVersion":"v1","time":"2023-06-11T23:17:12Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:labels":{".":{},"f:app":{},"f:app.kubernetes.io/managed-by":{},"f:app.kubernetes.io/name":{},"f:scylla-operator.scylladb.com/scylla-service-type":{},"f:scylla/cluster":{},"f:scylla/datacenter":{},"f:scylla/decommissioned":{},"f:scylla/rack":{},\
    "f:scylla/replace":{}}}}}]}}}                                                                                                  
...
2023-06-11 23:17:22,583 f:__init__.py     l:793  c:sdcm.utils.k8s       p:DEBUG > Processing following line: \
  {"type":"DELETED","object":{"kind":"Endpoints","apiVersion":"v1","metadata":{ \
    "name":"sct-cluster-us-east1-b-us-east1-2","namespace":"scylla","uid":"ab9b398c-da9e-4ae2-b154-84410f42ad25","resourceVersion":"42628","creationTimestamp":"2023-06-11T23:10:18Z","labels":{"app":"scylla","app.kubernetes.io/managed-by":"scylla-operator","app.kubernetes.io/name":"scylla","scylla-operator.scylladb.com/scylla-service-type":"member","scylla/cluster":"sct-cluster","scylla/datacenter":"us-east1-b","scylla/decommissioned":"true","scylla/rack":"us-east1","scylla/replace":"172.20.26.177"},"managedFields":[{"manager":"kube-controller-manager","operation":"Update","apiVersion":"v1","time":"2023-06-11T23:17:12Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:labels":{".":{},"f:app":{},"f:app.kubernetes.io/managed-by":{},"f:app.kubernetes.io/name":{},"f:scylla-operator.scylladb.com/scylla-service-type":{},"f:scylla/cluster":{},"f:scylla/datacenter":{},"f:scylla/decommissioned":{},"f:scylla/rack":{}, \
    "f:scylla/replace":{}}}}}]}}}
```

And scylla-operator failure:
```
E0611 23:17:22.594024       1 runtime/runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 513 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1dae300?, 0x34ba840})
	k8s.io/apimachinery@v0.27.1/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0015fb4b0?})
	k8s.io/apimachinery@v0.27.1/pkg/util/runtime/runtime.go:49 +0x75
panic({0x1dae300, 0x34ba840})
	runtime/panic.go:884 +0x213
k8s.io/api/core/v1.(*Pod).GetNamespace(0x20a295e?)
	<autogenerated>:1 +0x9
github.com/scylladb/scylla-operator/pkg/naming.ObjRef({0x24ed150, 0x0})
	github.com/scylladb/scylla-operator/pkg/naming/names.go:19 +0x2e
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).syncServices(0xc000052c80, {0x24d2e38, 0xc0004d4aa0}, 0xc0018fc2c0, 0xc00003a5b0, 0x20d492f?, 0xc001799b60?)
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/sync_services.go:369 +0x829
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).sync.func7()
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/sync.go:266 +0x35
github.com/scylladb/scylla-operator/pkg/controllerhelpers.RunSync(0xc00023a1c0?, {0x20cba40, 0x1c}, {0x20c5b26, 0x19}, 0xc, 0x0?)
	github.com/scylladb/scylla-operator/pkg/controllerhelpers/sync.go:102 +0x63
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).sync(0xc000052c80, {0x24d2e38, 0xc0004d4aa0}, {0xc0015a9560, 0x12})
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/sync.go:260 +0x2be5
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).processNextItem(0xc000052c80, {0x24d2e38, 0xc0004d4aa0})
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/controller.go:235 +0x12d
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).runWorker(...)
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/controller.go:259
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:259 +0x25
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x37335172342b674e?, {0x24b5540, 0xc000a8f950}, 0x1, 0xc0000feba0)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x66784f6368747130?, 0x3b9aca00, 0x0, 0x77?, 0x74567379544e6742?)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext({0x24d2e38, 0xc0004d4aa0}, 0xc0009197a8, 0x71575745774a6246?, 0x32482f6d67676838?, 0x4f?)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:259 +0x99
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(...)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:170
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).Run.func2()
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/controller.go:284 +0x8d
created by github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).Run
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/controller.go:282 +0x1b3
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xc0d769]

goroutine 513 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0015fb4b0?})
	k8s.io/apimachinery@v0.27.1/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x1dae300, 0x34ba840})
	runtime/panic.go:884 +0x213
k8s.io/api/core/v1.(*Pod).GetNamespace(0x20a295e?)
	<autogenerated>:1 +0x9
github.com/scylladb/scylla-operator/pkg/naming.ObjRef({0x24ed150, 0x0})
	github.com/scylladb/scylla-operator/pkg/naming/names.go:19 +0x2e
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).syncServices(0xc000052c80, {0x24d2e38, 0xc0004d4aa0}, 0xc0018fc2c0, 0xc00003a5b0, 0x20d492f?, 0xc001799b60?)
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/sync_services.go:369 +0x829
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).sync.func7()
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/sync.go:266 +0x35
github.com/scylladb/scylla-operator/pkg/controllerhelpers.RunSync(0xc00023a1c0?, {0x20cba40, 0x1c}, {0x20c5b26, 0x19}, 0xc, 0x0?)
	github.com/scylladb/scylla-operator/pkg/controllerhelpers/sync.go:102 +0x63
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).sync(0xc000052c80, {0x24d2e38, 0xc0004d4aa0}, {0xc0015a9560, 0x12})
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/sync.go:260 +0x2be5
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).processNextItem(0xc000052c80, {0x24d2e38, 0xc0004d4aa0})
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/controller.go:235 +0x12d
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).runWorker(...)
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/controller.go:259
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:259 +0x25
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x37335172342b674e?, {0x24b5540, 0xc000a8f950}, 0x1, 0xc0000feba0)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x66784f6368747130?, 0x3b9aca00, 0x0, 0x77?, 0x74567379544e6742?)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext({0x24d2e38, 0xc0004d4aa0}, 0xc0009197a8, 0x71575745774a6246?, 0x32482f6d67676838?, 0x4f?)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:259 +0x99
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(...)
	k8s.io/apimachinery@v0.27.1/pkg/util/wait/backoff.go:170
github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).Run.func2()
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/controller.go:284 +0x8d
created by github.com/scylladb/scylla-operator/pkg/controller/scyllacluster.(*Controller).Run
	github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/controller.go:282 +0x1b3
```

After it the scylla-operator gets restarted and running.
But the ongoing `replace Scylla member` procedure fails with the following error:

```
INFO  2023-06-11 23:17:10,339 [shard  0] storage_service - DECOMMISSIONING: set_bootstrap_state done
INFO  2023-06-11 23:17:10,340 [shard  0] storage_service - entering DECOMMISSIONED mode             
INFO  2023-06-11 23:17:10,340 [shard  0] storage_service - DECOMMISSIONING: done   
...
INFO  2023-06-11 23:17:14,698 [shard  0] init - Shutting down sighup was successful                 
INFO  2023-06-11 23:17:14,698 [shard  0] init - Scylla version 2022.2.6-0.20230423.b9a1478dba07 shutdown complete.
2023-06-11 23:17:14,730 INFO waiting for scylla to stop                                             
2023-06-11 23:17:14,834 INFO stopped: scylla (exit status 0)                                        
I0611 23:17:14.882339       1 signals/signal.go:26] Received shutdown signal: terminated; shutting down...
I0611 23:17:14.882456       1 operator/sidecar.go:431] "Waiting for scylla process to finish"       
I0611 23:17:14.882497       1 operator/sidecar.go:447] "Sending SIGTERM to the scylla process"      
I0611 23:17:14.882500       1 operator/sidecar.go:406] "Shutting down Prober server"                
I0611 23:17:14.882530       1 operator/sidecar.go:453] "Sent SIGTERM to the scylla process"         
I0611 23:17:14.882470       1 sidecar/controller.go:161] "Shutting down controller" Controller="SidecarController"
I0611 23:17:14.882568       1 operator/sidecar.go:399] "Prober server shut down"                    
I0611 23:17:14.882609       1 operator/sidecar.go:415] "Shut down Prober server"                    
I0611 23:17:14.882663       1 sidecar/controller.go:164] "Shut down controller" Controller="SidecarController"
2023-06-11 23:17:15,836 WARN received SIGTERM indicating exit request                               
2023-06-11 23:17:15,837 INFO waiting for rsyslog, scylla-housekeeping, scylla-jmx, scylla-node-exporter, sshd to die
2023-06-11 23:17:16,838 INFO stopped: sshd (terminated by SIGTERM)                                  
2023-06-11 23:17:16,840 INFO stopped: scylla-node-exporter (terminated by SIGTERM)                  
2023-06-11 23:17:17,162 INFO stopped: scylla-jmx (exit status 143)                                  
2023-06-11 23:17:18,164 INFO stopped: scylla-housekeeping (terminated by SIGTERM)                   
2023-06-11 23:17:18,165 INFO stopped: rsyslog (exit status 0)                                       
I0611 23:17:18.199284       1 operator/sidecar.go:438] "Scylla process finished"                    
I0611 23:19:37.835518       1 operator/sidecar.go:147] sidecar version "v1.9.0-beta.1-0-g203d34d"  
...
E0611 23:19:39.870518       1 sidecar/probes.go:154] "healthz probe: can't connect to Scylla API" err="dial tcp [::1]:10000: connect: connection refused" Service="scylla/sct-cluster-us-east1-b-us-east1-2"
INFO  2023-06-11 23:19:39,904 [shard  0] iotune - /var/lib/scylla/data passed sanity checks         
INFO  2023-06-11 23:19:39,904 [shard  0] iotune - Disk parameters: max_iodepth=2046 disks_per_array=2 minimum_io_size=512
Starting Evaluation. This may take a while...
...
Measuring random read IOPS: 831846 IOPS
...
2023-06-11 23:21:41,097 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
...
INFO  2023-06-11 23:21:43,235 [shard  0] init - Scylla version 2022.2.6-0.20230423.b9a1478dba07 with build-id d9def2e0e62ae9483d4becd80334de8b262894c4 starting ...
...
INFO  2023-06-11 23:21:43,644 [shard  0] system_keyspace - Setting local host id to 2ca4df46-8b3b-42e1-91cb-8e449b8780ca
...
INFO  2023-06-11 23:21:43,945 [shard  0] storage_service - Gathering node replacement information for 172.20.26.177
INFO  2023-06-11 23:21:43,946 [shard  0] storage_service - Checking remote features with gossip     
INFO  2023-06-11 23:21:43,946 [shard  0] gossip - Gossip shadow round started with nodes={172.20.61.221}
INFO  2023-06-11 23:21:43,947 [shard  0] gossip - Gossip shadow round finisehd with nodes_talked={172.20.61.221}
INFO  2023-06-11 23:21:43,948 [shard  0] gossip - Feature check passed. Local node 172.20.196.96 features = {AGGREGATE_STORAGE_OPTIONS, ALTERNATOR_STREAMS, ALTERNATOR_TTL, CDC, CDC_GENERATIONS_V2, COMPUTED_COLUMNS, CORRECT_COUNTER_ORDER, CORRECT_IDX_TOKEN_IN_SECONDARY_INDEX, CORRECT_NON_COMPOUND_RANGE_TOMBSTONES, CORRECT_STATIC_COMPACT_IN_MC, COUNTERS, DIGEST_FOR_NULL_VALUES, DIGEST_INSENSITIVE_TO_EXPIRY, DIGEST_MULTIPARTITION_READ, HINTED_HANDOFF_SEPARATE_CONNECTION, INDEXES, IN_MEMORY_TABLES, KEYSPACE_STORAGE_OPTIONS, LARGE_PARTITIONS, LA_SSTABLE_FORMAT, LWT, MATERIALIZED_VIEWS, MC_SSTABLE_FORMAT, MD_SSTABLE_FORMAT, ME_SSTABLE_FORMAT, NONFROZEN_UDTS, PARALLELIZED_AGGREGATION, PER_TABLE_CACHING, PER_TABLE_PARTITIONERS, RANGE_SCAN_DATA_VARIANT, RANGE_TOMBSTONES, ROLES, ROW_LEVEL_REPAIR, SCHEMA_COMMITLOG, SCHEMA_TABLES_V3, SEPARATE_PAGE_SIZE_AND_SAFETY_LIMIT, STREAM_WITH_RPC_STREAM, TOMBSTONE_GC_OPTIONS, TRUNCATION_TABLE, TYPED_ERRORS_IN_READ_RPC, UDA, UDA_NATIVE_PARALLELIZED_AGGREGATION, UNBOUNDED_RANGE_TOMBSTONES, VIEW_VIRTUAL_COLUMNS, WORKLOAD_PRIORITIZATION, WRITE_FAILURE_REPLY, XXHASH}, Remote common_features = {AGGREGATE_STORAGE_OPTIONS, ALTERNATOR_STREAMS, ALTERNATOR_TTL, CDC, CDC_GENERATIONS_V2, COMPUTED_COLUMNS, CORRECT_COUNTER_ORDER, CORRECT_IDX_TOKEN_IN_SECONDARY_INDEX, CORRECT_NON_COMPOUND_RANGE_TOMBSTONES, CORRECT_STATIC_COMPACT_IN_MC, COUNTERS, DIGEST_FOR_NULL_VALUES, DIGEST_INSENSITIVE_TO_EXPIRY, DIGEST_MULTIPARTITION_READ, HINTED_HANDOFF_SEPARATE_CONNECTION, INDEXES, IN_MEMORY_TABLES, KEYSPACE_STORAGE_OPTIONS, LARGE_PARTITIONS, LA_SSTABLE_FORMAT, LWT, MATERIALIZED_VIEWS, MC_SSTABLE_FORMAT, MD_SSTABLE_FORMAT, ME_SSTABLE_FORMAT, NONFROZEN_UDTS, PARALLELIZED_AGGREGATION, PER_TABLE_CACHING, PER_TABLE_PARTITIONERS, RANGE_SCAN_DATA_VARIANT, RANGE_TOMBSTONES, ROLES, ROW_LEVEL_REPAIR, SCHEMA_COMMITLOG, SCHEMA_TABLES_V3, SEPARATE_PAGE_SIZE_AND_SAFETY_LIMIT, STREAM_WITH_RPC_STREAM, TOMBSTONE_GC_OPTIONS, TRUNCATION_TABLE, TYPED_ERRORS_IN_READ_RPC, UDA, UDA_NATIVE_PARALLELIZED_AGGREGATION, UNBOUNDED_RANGE_TOMBSTONES, VIEW_VIRTUAL_COLUMNS, WORKLOAD_PRIORITIZATION, WRITE_FAILURE_REPLY, XXHASH}
INFO  2023-06-11 23:21:43,948 [shard  0] init - Shutting down group 0 service
...
INFO  2023-06-11 23:21:44,107 [shard  0] init - Shutting down sighup                                
INFO  2023-06-11 23:21:44,107 [shard  0] init - Shutting down sighup was successful  
ERROR 2023-06-11 23:21:44,107 [shard  0] init - Startup failed: std::runtime_error (Cannot replace_address 172.20.26.177 because it doesn't exist in gossip)
2023-06-11 23:21:44,214 INFO exited: scylla (exit status 1; not expected)                           
2023-06-11 23:21:45,217 INFO spawned: 'scylla' with pid 230                                         
Scylla version 2022.2.6-0.20230423.b9a1478dba07 with build-id d9def2e0e62ae9483d4becd80334de8b262894c4 starting ...
```

**To Reproduce**
Steps to reproduce the behavior:
1. Deploy scylla operator
2. Run in circles the `decommission and replace Scylla member` procedure
3. See error (in our case it was `5th` attempt)

**Expected behavior**
Scylla-operator should not crash and the `replace` procedure must succeed.

**Logs**
- **db-cluster-3dfcaf86.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/db-cluster-3dfcaf86.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/db-cluster-3dfcaf86.tar.gz)
- **sct-runner-events-3dfcaf86.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/sct-runner-events-3dfcaf86.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/sct-runner-events-3dfcaf86.tar.gz)
- **sct-3dfcaf86.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/sct-3dfcaf86.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/sct-3dfcaf86.log.tar.gz)
- **monitor-set-3dfcaf86.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/monitor-set-3dfcaf86.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/monitor-set-3dfcaf86.tar.gz)
- **loader-set-3dfcaf86.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/loader-set-3dfcaf86.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/loader-set-3dfcaf86.tar.gz)
- **kubernetes-3dfcaf86.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/kubernetes-3dfcaf86.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/3dfcaf86-8765-4012-b0bc-ffaab9bfb0b0/20230612_001651/kubernetes-3dfcaf86.tar.gz)

**Environment:**
 - Platform: EKS
 - Kubernetes version: v1.25.10-eks-c12679a
 - Scylla version: 2022.2.6-0.20230423.b9a1478dba07 with build-id d9def2e0e62ae9483d4becd80334de8b262894c4
 - Scylla-operator version: e.g.: 1.9.0-rc.0

--- a/pkg/controller/scyllacluster/sync_services.go
+++ b/pkg/controller/scyllacluster/sync_services.go
@@ -360,13 +360,13 @@ func (scc *Controller) syncServices(
 
 				klog.V(2).InfoS("Pod has not been recreated by the StatefulSet controller yet",
 					"ScyllaCluster", klog.KObj(sc),
-					"Pod", klog.KObj(pod),
+					"Pod", klog.KObj(svc),
 				)
 				progressingConditions = append(progressingConditions, metav1.Condition{
 					Type:               serviceControllerProgressingCondition,
 					Status:             metav1.ConditionTrue,
 					Reason:             "WaitingForPodRecreation",
-					Message:            fmt.Sprintf("Service %q is waiting for StatefulSet controller to recreate pod %q.", naming.ObjRef(svc), naming.ObjRef(pod)),
+					Message:            fmt.Sprintf("Service %q is waiting for StatefulSet controller to recreate pod.", naming.ObjRef(svc)),
 					ObservedGeneration: sc.Generation,
 				})
 			} else {

